mobility/planner_scp/scp/src/traj_opt.cpp

```diff
@@ -316,17 +856,34 @@ void TOP::UpdateF(Vec7& f, Vec13& X, Vec6& U) {
 void TOP::UpdateA(Mat7& A, Vec13& X, Vec6& U) {
   A.setZero();
 
+  decimal_t qx = X(6);
+  decimal_t qy = X(7);
+  decimal_t qz = X(8);
+  decimal_t qw = X(9);
+
   decimal_t wx = X(10);
   decimal_t wy = X(11);
   decimal_t wz = X(12);
 
-  Mat4 omega_skew;
-  omega_skew << 0, wz, -wy, wz,
-            -wz, 0, wx, wy,
-            wy, -wx, 0, wz,
-            -wx, -wy, -wz, 0;
-
-  A.block(0, 0, 4, 4) = omega_skew;
+  Mat4 df_dq;
+  df_dq << 0, -wz, wy, wx,
+                wz, 0, -wx, wy,
+                -wy, wx, 0, wz,
+                -wx, -wy, -wz, 0;
+  A.block(0, 0, 4, 4) = 0.5*df_dq;
+
+  // TODO(somrita): Remove below matrix
+  // omega_skew << 0, wz, -wy, wz,
+  //           -wz, 0, wx, wy,
+  //           wy, -wx, 0, wz,
+  //           -wx, -wy, -wz, 0;
+
+  Mat4x3 df_dw;
```

Comment:
`Mat4x3` doesn't seem to be defined in _types.h_ .

Reply:
Eagle eye! Yes, I forgot to copy over types.h. I added a commit to this branch.